Source/WebCore/platform/graphics/egl/GLFence.cpp

```diff
@@ -21,70 +21,87 @@
 #include "GLFence.h"
 
 #include "GLContext.h"
+#include "GLFenceEGL.h"
+#include "GLFenceGL.h"
 
 #if USE(LIBEPOXY)
 #include <epoxy/gl.h>
 #else
-#include <GLES3/gl3.h>
+#include <GLES2/gl2.h>
 #endif
 
 namespace WebCore {
 
-bool GLFence::isSupported()
+const GLFence::Capabilities& GLFence::capabilities()
 {
     static std::once_flag onceFlag;
-    static bool supported = false;
-
-    std::call_once(onceFlag, [&]() {
-        auto version = GLContext::versionFromString(reinterpret_cast<const char*>(glGetString(GL_VERSION)));
-        if (version >= 300) {
-            supported = true;
-            return;
+    static Capabilities capabilities;
+    std::call_once(onceFlag, [] {
+        auto& display = PlatformDisplay::sharedDisplay();
+        const auto& extensions = display.eglExtensions();
+        if (display.eglCheckVersion(1, 5)) {
+            capabilities.eglSupported = true;
+            capabilities.eglServerWaitSupported = true;
+        } else {
+            capabilities.eglSupported = extensions.KHR_fence_sync;
+            capabilities.eglServerWaitSupported = extensions.KHR_fence_sync;
         }
-
-        const char* extensionsString = reinterpret_cast<const char*>(glGetString(GL_EXTENSIONS));
-        if (GLContext::isExtensionSupported(extensionsString, "GL_APPLE_sync"))
-            supported = true;
+#if OS(LINUX)
+        capabilities.eglExportableSupported = extensions.ANDROID_native_fence_sync;
+#endif
+        capabilities.glSupported = GLContext::versionFromString(reinterpret_cast<const char*>(glGetString(GL_VERSION))) >= 300;
```

Comment:
We need gl fence support in order to have egl fence support, so the values of eglSupported and eglServerWaitSupported should depend on the value of glSupported.

Reply:
I assume this is implicit, if driver exposes egl fence extensions is because we can use the api, no? If it fails later at runtime to create the fence it will have the same effect.